Mathlib/CategoryTheory/Limits/IsConnected.lean

```diff
@@ -137,4 +137,37 @@ theorem isConnected_iff_of_initial (F : C ⥤ D) [F.Initial] : IsConnected C ↔
 
 end Functor
 
+section
+
+variable (C : Type*) [Category C]
+
+/-- Prove that a category is connected by supplying an explicit initial object. -/
+lemma isConnected_of_isInitial {x : C} (h : Limits.IsInitial x) : IsConnected C := by
+  letI : Nonempty C := ⟨x⟩
+  apply isConnected_of_zigzag
+  intro j₁ j₂
+  use [x, j₂]
+  simp only [List.chain_cons, List.Chain.nil, and_true, ne_eq, reduceCtorEq, not_false_eq_true,
+    List.getLast_cons, List.cons_ne_self, List.getLast_singleton]
+  exact ⟨Zag.symm <| Zag.of_hom <| h.to _, Zag.of_hom <| h.to _⟩
+
+/-- Prove that a category is connected by supplying an explicit terminal object. -/
+lemma isConnected_of_isTerminal {x : C} (h : Limits.IsTerminal x) : IsConnected C := by
+  letI : Nonempty C := ⟨x⟩
+  apply isConnected_of_zigzag
+  intro j₁ j₂
+  use [x, j₂]
+  simp only [List.chain_cons, List.Chain.nil, and_true, ne_eq, reduceCtorEq, not_false_eq_true,
+    List.getLast_cons, List.cons_ne_self, List.getLast_singleton]
+  exact ⟨Zag.of_hom <| h.from _, Zag.symm <| Zag.of_hom <| h.from _⟩
+
+-- note : it seems making the following two as instances breaks things, so these are lemmas.
+lemma isConnected_of_hasInitial [Limits.HasInitial C] : IsConnected C :=
+  isConnected_of_isInitial C (Limits.initialIsInitial)
```

Comment:
```suggestion
  isConnected_of_isInitial C Limits.initialIsInitial
```
and similar below

Reply:
My addiction to parentheses again... I was a bit too late to see it, sorry. I made #23443 with the fix.